internal/hcp/internal/controllers/link/controller.go

```diff
@@ -118,7 +120,14 @@ func (r *linkReconciler) Reconcile(ctx context.Context, rt controller.Runtime, r
 		return writeStatusIfNotEqual(ctx, rt, res, newStatus)
 	}
 
-	hcpClient, err := r.hcpClientFn(&link)
+	existingCfg := r.hcpManager.GetCloudConfig()
+	newCfg := config.CloudConfig{
+		ResourceID:   link.ResourceId,
+		ClientID:     link.ClientId,
+		ClientSecret: link.ClientSecret,
+	}
+	cfg := config.Merge(existingCfg, newCfg)
```

Comment:
Can we clarify why we need to merge instead of changing the `existingCfg` values directly?

Reply:
I'll add a comment about this! The reasoning is:
1. Some values can be set at startup like what HCP API endpoint to use. We don't currently have a way in the linking API to set this, but support for this will be added later, so I thought a merge made more sense in the long term.
2. The NodeID and the NodeName of the cluster is set on the cloud config when Consul starts, so we want to continue to use these values. I considered some different strategies around this like passing these two values as dependencies to the controller, but I figured I'll want to implement a merge anyway because of (1).